rhel/vex/parser.go

```diff
@@ -0,0 +1,450 @@
+	c                  *csaf.CSAF
+	pc                 *productCache
+	rc                 *repoCache
+}
+
+func (c *creator) knownAffectedVulnerabilities(ctx context.Context, v csaf.Vulnerability, protoVulnFunc func() *claircore.Vulnerability) ([]*claircore.Vulnerability, error) {
+	// Known affected
+	out := []*claircore.Vulnerability{}
+	for _, pc := range v.ProductStatus["known_affected"] {
+		rel := c.c.ProductTree.Relationships.FindRelationship(pc, "default_component_of")
+		if rel == nil {
+			// Should never get here, error in data
+			zlog.Debug(ctx).
+				Str("product:package", pc).
+				Msg("could not find a relationship for product:package")
+			continue
+		}
+		if strings.HasPrefix(rel.ProductRef, "kernel") {
+			// We don't want to ingest kernel advisories as
+			// containers have no say in the kernel.
+			continue
+		}
+		repoProd := c.pc.Get(c.c, rel.RelatesToProductRef)
+		if repoProd == nil {
+			// Should never get here, error in data
+			zlog.Warn(ctx).
+				Str("prod", rel.RelatesToProductRef).
+				Msg("could not find product in product tree")
+			continue
+		}
+		cpeHelper, ok := repoProd.IdentificationHelper["cpe"]
+		if !ok {
+			zlog.Warn(ctx).
+				Str("prod", rel.RelatesToProductRef).
+				Msg("could not find cpe helper type in product")
+			continue
+		}
+
+		// pkgName will be overriden if we find a valid pURL
+		pkgName := rel.ProductRef
+		compProd := c.pc.Get(c.c, rel.ProductRef)
+		if compProd == nil {
+			// Should never get here, error in data
+			zlog.Warn(ctx).
+				Str("pkg", rel.ProductRef).
+				Msg("could not find package in product tree")
+			continue
+		}
+		// It is possible that we will not find a pURL, in that case
+		// the package.Name will be reported as-is.
+		purlHelper, ok := compProd.IdentificationHelper["purl"]
+		if ok {
+			purl, err := packageurl.FromString(purlHelper)
+			switch {
+			case err != nil:
+				zlog.Warn(ctx).
+					Str("purl", purlHelper).
+					Err(err).
+					Msg("could not parse PURL")
+			default:
+				pkgName = purl.Name
+			}
+			if purl.Type != packageurl.TypeRPM || purl.Namespace != "redhat" {
+				// Just ingest advisories that are Red Hat RPMs, this will
+				// probably change down the line when we consolidate updaters.
+				continue
+			}
+		}
+		if strings.HasPrefix(pkgName, "kernel") {
+			// We don't want to ingest kernel advisories as
+			// containers have no say in the kernel.
+			continue
+		}
+
+		vuln := protoVulnFunc()
+		// What is the deal here? Just stick the package name in and f-it?
+		// That's the plan so far as there's no PURL product ID helper.
+		vuln.Package = &claircore.Package{
+			Name: pkgName,
+			Kind: claircore.SOURCE,
+		}
+
+		wfn, err := cpe.Unbind(cpeHelper)
+		if err != nil {
+			return nil, fmt.Errorf("could not unbind cpe: %s %w", cpeHelper, err)
+		}
+		vuln.Repo = c.rc.Get(wfn)
+		if sc := c.c.FindScore(pc); sc != nil {
+			vuln.NormalizedSeverity, vuln.Severity, err = CVSSVectorFromScore(sc)
```

Comment:
This should be related to the "threats" field for the CPE instead of the CVSS score

Reply:
We want to keep the `vuln.Severity` as the CVSS score though right (so we are able to surface it to the user)? Is there some doc that says we should use the `threats` field? (not saying we shouldn't, it'd just be nice to reference something and explain why the `vuln.NormalizedSeverity` and `vuln.Severity` might equate to different normalized values.